### PR TITLE
Dev/issue 9970 job schema fix

### PR DESCRIPTION
**Pull request by mcantelon**
Added a new taxonomy, and corresponding terms, for the three job statuses (completed, in progress, and halted due to an error). Make the statusId property of the QubitJob model a foreign key to QubitTerm. Added migration to make these changes on old versions of the database.

### CODE REVIEW

File: data/fixtures/settings.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ QubitSetting:
     name: version
     editable: 0
     deleteable: 0
-    value: 149
+    value: 150
```
Comment (sevein) [inline data/fixtures/settings.yml]: This should be moved to 152.

Comment (sevein) [reply data/fixtures/settings.yml]: Related: #515 and #541.

**Review by sevein — CHANGES_REQUESTED** [review]
@mcantelon, this is looking great. Glad we're fixing this.

You may have noticed that previous PRs did not bump the db version in the fixture. You have a chance to solve that here if you want. Related: #515 and #541.

File: data/fixtures/settings.yml (hunk 1 above)
Comment (mcantelon) [reply data/fixtures/settings.yml]: Ah, I didn't notice that the bumped number was less than my migration. Will fix. Thanks!

**Comment by sbreker** [comment]
Thanks Mike.  Good catch Sevein!  I completely forgot about the version
setting.

**Pull request merged**